plugin/trino-iceberg/src/main/java/io/trino/plugin/iceberg/catalog/hms/HiveMetastoreTableOperations.java

```diff
@@ -88,21 +88,15 @@ protected void commitToExistingTable(TableMetadata base, TableMetadata metadata)
                         .setParameter(METADATA_LOCATION_PROP, newMetadataLocation)
                         .setParameter(PREVIOUS_METADATA_LOCATION_PROP, currentMetadataLocation)
                         .build();
+
+                // todo privileges should not be replaced for an alter
+                PrincipalPrivileges privileges = table.getOwner().map(MetastoreUtil::buildInitialPrivilegeSet).orElse(NO_PRIVILEGES);
+                metastore.replaceTable(database, tableName, table, privileges);
             }
             catch (RuntimeException e) {
-                try {
-                    io().deleteFile(newMetadataLocation);
```

Comment:
This usecase is probably pretty exotic to catch in a real-world test case.
Would it make sense to have a unit test case to simulate such a situation?

Reply:
I can try to unit test this but in my opinion it wouldn't be very useful, also although code here is pretty simple it would require a lot of mocking and it is only a little part of whole commit process that is initiated in `org.apache.iceberg.BaseTransaction#commitTransaction` so testing this small piece wouldn't be meaningful.